sdk/eventhub/event-hubs/src/eventHubReceiver.ts

```diff
@@ -608,10 +739,14 @@ export class EventHubReceiver extends LinkEntity {
           this.name,
           options
         );
-        // It is possible for someone to close the receiver and then start it again.
-        // Thus make sure that the receiver is present in the client cache.
-        if (!this._context.receivers[this.name]) this._context.receivers[this.name] = this;
+        // store the underlying link in a cache
+        this._context.receivers[this.name] = this;
+
         await this._ensureTokenRenewal();
+        if (this._onMessage) {
+          // add at least 1 credit when the user has a registered handler
+          this._addCredit(this.prefetchCount || 1);
```

Comment:
We seem to have multiple places where we are adding credits, I'd like to simplify this if possible. Here are my thoughts
- `registerHandlers` adds the right amount of credits for both streaming and batching receivers
      - For streaming, its always the `defaultPrefetchCount` because we are always creating a receiver link from scratch
      - For batching, it is the logic you already have in place
- `intialize()` should not add any credits
- `detached()` should add credits after `initialize()` is done only if the current mode is streaming. The amount to add is `defaultPrefetchCount`
- `onAmqpMessage` should not add any credits

Thoughts?